troubleshooting/tools/log-loss-test-framework/logger/log_generator.c

```diff
@@ -0,0 +1,153 @@
+#define ONE_KB_TEXT "GRHGCNRDCTLUWQCBFBKFGZHTRUDQEWDDKBVMHPYVOAHGADVQGRHGCNRDCTLUWQCBFBKFGZHTGEUKFXWNCKXPRWBSVJGHEARMDQGVVRFPVCIBYEORHYPUTQJKUMNZJXIYLDCJUHABJIXFPUNJQDORGPKWFLQZXIGVGCWTZCVWGBFSGVXGEITYKNTWCYZDOAZFOTXDOFRPECXBSCSORSUUNUJZEJZPTODHBXVMOETBRFGNWNZHGINVNYZPKKSFLZHLSSDHFGLTHZEKICPGNYSCTAIHARDDYIJHKLMAOIDLEKRXMFNVJOJVDFYKNVIQKCIGTRFWKJRHQSFDWWKTJNMNKFBOMBMZMRCOHPUFZEPTQTZBLBDBZPJJXRYDFSOWKDVZLZYWSJYFTCKQJFPQOMCWQHKLNHUGWWVBGTRLLVUHTPHTKNBSRUNNOIFGIJPBHPCKYXNGDCQYJEWFFKRRTHJDUBEZPJIXMAOLZQDZQAYEUZFRLTLTXNGAVAGZZDUERZWTJVDTXPKOIRTCKTFOFJAXVFLNKPBYOIYVPHUYBRZZORCEMMAUTZIAUSXVDTKHSUIRTSYWQMYZBMUGSATXPNESEVQMUKHYZFWSLHJDNYUQWOKDUTUKPRXBLIYGSCFGBGXATINMMCWNWBGJTLZTPKGBTPWTHQPUHDJITWPCJLGZFNZTCIEWWVTREFCTPVOUADQCRQCBRHNHDKGQIXHIWGGDGAAFYZRODKFTKQATAUDOMZTSQUYZHGNJOBSUJDHESPBOIJCGXPEZMMQJNFTYBJEYXPZAZICZJKEZKCZEUMZTTSQEHADOVMCDMDEBUJAPKIAEYQEWIYZSAYAWAGFSTBJYCUFZHMJMLCTVTZWGCPDAURQYSXVICLVWKPAOMVTQTESYFPTMNMSNZPUXMDJRDKHDRAIRYELEXRJUAMOLZVWNHGNVFETVUDZEIDJRPSHMXAZDZXDCXMUJTPDTDUHBAZGPIQOUNUHMVLCZCSUUHGTE"
+
+long long timeInMilliseconds(void) {
+    struct timeval tv;
+
+    gettimeofday(&tv,NULL);
+    return (((long long)tv.tv_sec)*1000)+(tv.tv_usec/1000);
+}
+
+/* msleep(): Sleep for the requested number of milliseconds. */
+void msleep(long msec)
+{
+    struct timespec ts;
+    int res;
+
+    if (msec < 0)
+    {
+        errno = EINVAL;
+    }
+
+    ts.tv_sec = msec / 1000;
+    ts.tv_nsec = (msec % 1000) * 1000000;
+
+    do {
+        res = nanosleep(&ts, &ts);
+    } while (res && errno == EINTR);
+}
+
+int main()
+{
+    int sizeInKb = atoi(getenv("SIZE_IN_KB"));
+    int totalSizeInKb = atoi(getenv("TOTAL_SIZE_IN_MB")) * 1000;
+    int throughputInKb = atoi(getenv("THROUGHPUT_IN_KB"));
+    int cycleTimeInS = atoi(getenv("CYCLE_TIME_IN_SECONDS"));
+    int cycleTimeInMs = cycleTimeInS * 1000;
+    int j = 0;
+    long long startSeconds;
+    long long endSeconds;
+    long long timeDiff;
+    long long loggingStart;
+    long long loggingEnd;
+    long long actualTimeElapsed = 0;
+    long long expectedTimeElapsed = 0;
+
+    int idCounter = 10000000;
+    char* data = malloc(strlen(ONE_KB_TEXT) * sizeInKb + 1);
+    if (!data) {
+        printf("malloc failure");
+        return 1;
+    }
+
+    int totalMessages = totalSizeInKb / sizeInKb;
+    int messagesSent = 0;
+    int messagesPerCycle = (throughputInKb * cycleTimeInS) / sizeInKb;
+    int messagesThisCycle = 0;
+
+    char *burst_enabled = getenv("BURST_SIZE_IN_MB");
+    int burstSizeInMB = 0;
+    int burstThroughputInKb = 0;
+    int burstMessagesPerCycle = 0;
+    int totalBurstMessages = 0;
+    int burstDelayInMB = 0;
+    int burstDelayMessages = 0;
+    if (burst_enabled != NULL) {
+        burstSizeInMB = atoi(burst_enabled);
+        burstThroughputInKb = atoi(getenv("BURST_THROUGHPUT_IN_KB"));
+        if ((burstSizeInMB * 2) > totalSizeInKb) {
+            printf("ERROR: BURST_SIZE_IN_MB must be less than half of TOTAL_SIZE_IN_MB");
```

Comment:
The condition doesn't check what this print statement says. The print statement should say:
ERROR: BURST_SIZE_IN_MB must be greater than half of TOTAL_SIZE_IN_KB

Reply:
oh oops my math is wrong

Reply:
thanks

Reply:
Yep!